src/tasks/minions/cutLeapingFishActivity.ts

```diff
@@ -0,0 +1,98 @@
+import { percentChance } from 'e';
+import { Bank } from 'oldschooljs';
+
+import LeapingFish from '../../lib/skilling/skills/cooking/leapingFish';
+import { SkillsEnum } from '../../lib/skilling/types';
+import { CutLeapingFishActivityTaskOptions } from '../../lib/types/minions';
+import { handleTripFinish } from '../../lib/util/handleTripFinish';
+
+export const cutLeapingFishTask: MinionTask = {
+	type: 'CutLeapingFish',
+	async run(data: CutLeapingFishActivityTaskOptions) {
+		let { fishName, userID, channelID, quantity, duration } = data;
+		const user = await mUserFetch(userID);
+		const BarbarianFish = LeapingFish.find(LeapingFish => LeapingFish.name === fishName)!;
+
+		const currentLevel = user.skillLevel(SkillsEnum.Cooking);
+		let caviarChance = 0;
+		let caviarCreated = 0;
+		let roeChance = 0;
+		let roeCreated = 0;
+		let fishOffcutsChance = 0;
+		let fishOffcutsCreated = 0;
+
+		if (BarbarianFish.name === 'Cut leaping sturgeon') {
+			caviarChance = 1.25 * currentLevel;
+			fishOffcutsChance = (5 / 6) * 100;
+			if (caviarChance > 100) caviarChance = 100;
+			for (let i = 0; i < quantity; i++) {
+				if (percentChance(caviarChance)) {
+					caviarCreated += 1;
+				}
+			}
+			for (let i = 0; i < caviarCreated; i++) {
+				if (percentChance(fishOffcutsChance)) {
+					fishOffcutsCreated += 1;
+				}
+			}
+		}
+
+		if (BarbarianFish.name === 'Cut leaping salmon') {
+			roeChance = 1.25 * currentLevel;
+			fishOffcutsChance = (3 / 4) * 100;
+			if (roeChance > 100) roeChance = 100;
```

Comment:
Line 41 and 43 can be condensed to `roeChance = clamp(1.25 * currentLevel, 0, 100)`